Add Warning in when Wise is not Configured when Displaying Bank Details

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/4523

![image](https://user-images.githubusercontent.com/12435965/129282118-ad3160fa-6829-424f-a985-7909cd7055da.png)

znarf reviewed (changes requested):
The message is not correct as we're not talking about "this host" but about the Transferwise account that we're using platform wide.

You should then also check `slug ===TW_API_COLLECTIVE_SLUG` before displaying the message.

SudharakaP commented:
> The message is not correct as we're not talking about "this host" but about the Transferwise account that we're using platform wide.
> 
> You should then also check `slug ===TW_API_COLLECTIVE_SLUG` before displaying the message.

@znarf : I've changed the warning message a bit. But my understanding is that we first check for the hosts Wise account first and if it fails then we check for the platform-wide Wise account. Note that; 

https://github.com/opencollective/opencollective-frontend/blob/3ba540901793393e401dd8f23fd1c8e8f3498b6d/components/expenses/PayoutBankInformationForm.js#L352-L357

here we first take the slug of the `host` if it has Wise, otherwise we use the env variable. 

Furthermore I am not sure what you meant by `slug ===TW_API_COLLECTIVE_SLUG`. Could you please explain which `slug` you are talking about? The reason I added the condition, 

https://github.com/opencollective/opencollective-frontend/blob/3ba540901793393e401dd8f23fd1c8e8f3498b6d/components/expenses/PayoutBankInformationForm.js#L365

is that first we check if the host has Wise configured (in `host?.transferwise`) and then we check if our preceding query returned a value for `data?.host?.transferwise`, in which case we know that either the host has Wise or the collective pointed by `TW_API_COLLECTIVE_SLUG` is configured for Wise. 🤔

znarf commented:
Please double check the usage, in the case you're working on `host` is not an "host", it's the platform wide account.

<img width="596" alt="Screenshot 2021-08-13 at 21 02 30" src="https://user-images.githubusercontent.com/806/129407021-8182181c-1e27-4b11-8902-bb9283095696.png">

<img width="721" alt="Screenshot 2021-08-13 at 21 02 56" src="https://user-images.githubusercontent.com/806/129407041-6222a4d8-b2b2-4e2e-90f2-4d51e7f6e15d.png">

SudharakaP commented:
> Please double check the usage, in the case you're working on `host` is not an "host", it's the platform wide account.
> 
> <img alt="Screenshot 2021-08-13 at 21 02 30" width="596" src="https://user-images.githubusercontent.com/806/129407021-8182181c-1e27-4b11-8902-bb9283095696.png">
> 
> <img alt="Screenshot 2021-08-13 at 21 02 56" width="721" src="https://user-images.githubusercontent.com/806/129407041-6222a4d8-b2b2-4e2e-90f2-4d51e7f6e15d.png">

Oh, I think now it makes sense. Thanks. 👍🏽 I've went ahead and changed the condition to include `host.slug === TW_API_COLLECTIVE_SLUG && !data?.host?.transferwise`, so that we only go to this branch when the host is the platform wide one. Let me know if you see any issue.

kewitz commented on components/expenses/PayoutBankInformationForm.js:
```diff
@@ -362,6 +362,15 @@ const PayoutBankInformationForm = ({ isNew, getFieldName, host, fixedCurrency, i
+        <FormattedMessage
+          id="PayoutBankInformationForm.Warning.WiseNotConfigured"
+          defaultMessage="Wise is not configured on the platform."
+        />
```
I don't think you need to internationalize since this happens only to poorly configurated environments.

SudharakaP replied:
Makes sense. I've removed the internationalization for this. Thanks. 👍🏽

kewitz commented on components/expenses/PayoutBankInformationForm.js:
```diff
@@ -362,6 +362,15 @@ const PayoutBankInformationForm = ({ isNew, getFieldName, host, fixedCurrency, i
   // Display spinner if loading
   if (loading) {
     return <StyledSpinner />;
+  } else if (host.slug === TW_API_COLLECTIVE_SLUG && !data?.host?.transferwise) {
```
I think testing for `host.transferwise` should be enough due to the line `host = data?.host || host`.
```suggestion
  } else if (!host?.transferwise) {
```

SudharakaP replied:
Thanks, I've simplified the condition as you suggested. Seems to work fine. 👍🏽

znarf commented on components/expenses/PayoutBankInformationForm.js:
```diff
@@ -362,13 +362,10 @@ const PayoutBankInformationForm = ({ isNew, getFieldName, host, fixedCurrency, i
   if (loading) {
     return <StyledSpinner />;
-  } else if (host.slug === TW_API_COLLECTIVE_SLUG && !data?.host?.transferwise) {
+  } else if (!host?.transferwise) {
```
What's the rationale removing the previous restriction to `host.slug === TW_API_COLLECTIVE_SLUG` ?

SudharakaP replied:
Ops sorry, this is me being a bit confused. We need both of these. I've added it back. Should have been more careful. 🤯

znarf reviewed (approved):
Looks good to me.

znarf commented:
No one was supposed to see this and now it's breaking the experience of users in production.

Reverting.
